cloudpathlib/azure/azblobclient.py

```diff
@@ -202,20 +207,29 @@ def _check_hns(self, cloud_path: AzureBlobPath) -> Optional[bool]:
             try:
                 account_info = self.service_client.get_account_information()  # type: ignore
                 self._hns_enabled = account_info.get("is_hns_enabled", False)  # type: ignore
+
+            # get_account_information() not supported with this credential; we have to fallback to
+            # checking if the root directory exists and is a has 'metadata': {'hdi_isfolder': 'true'}
             except ResourceNotFoundError:
-                # get_account_information() not supported with this credential; we have to fallback to
-                # checking if the root directory exists and is a has 'metadata': {'hdi_isfolder': 'true'}
-                root_dir = self.service_client.get_blob_client(
-                    container=cloud_path.container, blob="/"
-                )
-                self._hns_enabled = (
-                    root_dir.exists()
-                    and root_dir.get_blob_properties().metadata.get("hdi_isfolder", False)
-                    == "true"
-                )
+                return self._check_hns_root_metadata(cloud_path)
+            except HttpResponseError as error:
+                if error.status_code == HTTPStatus.FORBIDDEN:
+                    self._check_hns_root_metadata(cloud_path)
```

Comment:
nit: to be consistent, either both of these should be a `return` statement or none of them should.